sdks/java/core/src/test/java/org/apache/beam/sdk/testing/PAssertTest.java

```diff
@@ -386,7 +386,7 @@ private static Throwable runExpectingAssertionFailure(Pipeline pipeline) {
     // is first caught by JUnit and causes a test failure.
     try {
       pipeline.run();
-    } catch (AssertionError exc) {
+    } catch (Exception exc) {
```

Comment:
The Spark runner will propagate the original error wrapped in a `RuntimeException`, see [here](https://github.com/apache/incubator-beam/blob/master/runners/spark/src/main/java/org/apache/beam/runners/spark/SparkRunner.java#L172).

I could change this to catching a `Throwable`, which is also the return value of whatever is caught anyway. Also note that the message assertion passes so it does surface the original error.

WDYT ?